File: Data Structures & Algorithms/missing-number/submission-5.java
class Solution {
    public int missingNumber(int[] nums) {
        int local = 0;
        int total = 0;

        for (int i : nums) {
            local = local ^ i;
        }

        for (int i = 0; i <= nums.length; i++) {
            total = total ^ i;
        }

        return local ^ total;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-5.py
import math

class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        maxBan = max(piles)
        left, right = 1, maxBan

        while left <= right:
            mid = (left + right) // 2
            local = 0
            for i in piles:
                local += math.ceil(i / mid)
            if local > h:
                left = mid + 1
            else:
                right = mid - 1
        return left
